src/amplifyp/gui/views/input/primer_row.py

```
@@ -82,6 +91,54 @@ def __init__(
+        if seq.strip():
+            try:
+                from amplifyp.dna import Primer
+                from amplifyp.gui.util import clean_sequence
+
+                cleaned_seq = clean_sequence(seq)
+                if cleaned_seq:
+                    primer_obj = Primer(sequence=cleaned_seq, name=name)
+                    tm = self.settings.calculate_primer_tm(primer_obj)
+                    self._tm_value = tm
+                    tm_val = f"{tm:.1f}°C"
+            except Exception:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)